### PR TITLE
i18n(ja): Add new `cloudflare.mdx`  translation

### PR DESCRIPTION
**Pull request by piro-hiroki**
#### Description (required)

Translate `cloudflare.mdx` into Japanese！

file: [guides/deploy/cloudflare.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/guides/deploy/cloudflare.mdx)

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! Just checking if this PR should still be marked as a draft?

If it's ready for editing, then please change the status so that it is ready for review! We are hoping to merge as many PRs by next week as we can before the docs site code migrates to Starlight. So if this is ready, we would like to be able to review it.  Thanks!

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! Is this one ready for a review now?

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    });
+    ```
+
+### 2つのモード
+
+現在、[`@astrojs/cloudflare`](https://github.com/withastro/astro/tree/main/packages/integrations/cloudflare#readme)アダプターでPages Functionsを使用する場合、2つのモードがサポートされています。
+
+1. **アドバンス**モード： このモードは `dist` 内の `_worker.js` をピックアップする `advanced` モードや、プロジェクトルートにあるfunctionsフォルダからWorkerをコンパイルするディレクトリモードで関数を実行したい場合に使用します。
+
+    モードが設定されていない場合、デフォルトは `"advanced"` です。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:115]:
```suggestion
    モードが設定されていない場合、デフォルトは`"advanced"`です。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+2. **ディレクトリ**モード： つまり、アダプタはアプリのクライアントサイド部分を同じようにコンパイルしますが、ワーカースクリプトをプロジェクトルートの `functions` フォルダに移動します。アダプタはそのフォルダに `[[path]].js` を置くだけなので、プラグインやページのミドルウェアを追加して、バージョン管理でチェックすることができます。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:117]:
```suggestion
2. **ディレクトリ**モード： 関数を`directory`モードで実行したい場合にこのモードを使用します。つまり、アダプタはアプリのクライアントサイド部分を同じようにコンパイルしますが、ワーカースクリプトをプロジェクトルートの`functions`フォルダに移動します。アダプタはそのフォルダに`[[path]].js`を置くだけなので、プラグインやページのミドルウェアを追加して、バージョン管理でチェックすることができます。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    ```ts title="astro.config.mjs" "directory"
+    export default defineConfig({
+      adapter: cloudflare({ mode: "directory" }),
+    });
+    ```
+### ページ関数の使用
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:124]:
```suggestion
### Pages Functionsの使用
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+[ページ・ファンクション](https://developers.cloudflare.com/pages/platform/functions/)は、サーバーサイドのコードを実行することで、専用のサーバーを実行することなく動的な機能を実現します。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:126]:
```suggestion
[Pages Functions](https://developers.cloudflare.com/pages/platform/functions/)により、専用のサーバーを実行することなしに、サーバーサイドのコードを実行して動的な機能を実現できます。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+始めるには、プロジェクトのルートに `/functions` ディレクトリを作成します。このディレクトリにFunctionsファイルを書くと、あらかじめ指定されたルートにカスタム機能を持ったWorkerが自動的に生成されます。Functionsの書き方については、[Pages Functions documentation](https://developers.cloudflare.com/pages/platform/functions/)を参照してください。
+
+📚 [SSR in Astro](/ja/guides/server-side-rendering/)についてもっと読む。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:130]:
```suggestion
📚 [AstroでのSSR](/ja/guides/server-side-rendering/)についてもっと読む。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+## トラブルシューティング
+
+エラーが発生する場合は、ローカルで使用している `node` のバージョン（`node -v`）が環境変数で指定しているバージョンと一致していることを再確認してください。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:134]:
```suggestion
エラーが発生する場合は、ローカルで使用している`node`のバージョン（`node -v`）が環境変数で指定しているバージョンと一致しているかどうかを再確認してください。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+Cloudflareは[Node v16.13](https://miniflare.dev/get-started/cli#installation)を必要とします。このバージョンはAstroのアウトオブボックス最小値よりも新しいので、少なくともv16.13を使用していることを再確認してください。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:136]:
```suggestion
Cloudflareは[Node v16.13](https://miniflare.dev/get-started/cli#installation)を必要とします。このバージョンはAstroが必要とする最小値よりも新しいので、少なくともv16.13を使用していることを再確認してください。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+CloudflareのAuto Minify設定の結果、クライアントサイドのハイドレーションが失敗することがあります。コンソールに `Hydration completed but contains mismatches` と表示されたら、Cloudflare settingsのAuto Minifyを無効にしてください。
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:138]:
```suggestion
CloudflareのAuto Minify設定の結果、クライアントサイドのハイドレーションが失敗することがあります。コンソールに`Hydration completed but contains mismatches`と表示されたら、Cloudflare settingsのAuto Minifyを無効にしてください。
```

File: src/content/docs/ja/guides/deploy/cloudflare.mdx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+アセットがアップロードされると、Wranglerはサイトを確認するためのプレビューURLを提供します。Cloudflare Pagesダッシュボードにログインすると、新しいプロジェクトが表示されます。
+
+### Wranglerでローカルにプレビューを有効にする
+
+プレビューを機能させるには、`wrangler` をインストールする必要がある。
+
+```bash
+pnpm install wrangler --save-dev
```
Comment (morinokami) [inline src/content/docs/ja/guides/deploy/cloudflare.mdx:63]: https://github.com/withastro/docs/pull/5284

```suggestion
pnpm add wrangler --save-dev
```

**Review by morinokami — CHANGES_REQUESTED** [review]
@sarah11918 Added some suggestions! I think we can merge this PR once those suggestions are merged.

@piro-hiroki Please let me know if you have anything to add!

**Review by sarah11918 — APPROVED** [review]
:rocket:

**Pull request merged**